app/src/main/java/org/mozilla/fenix/library/bookmarks/BookmarkFragment.kt

```diff
@@ -389,4 +391,17 @@ class BookmarkFragment : LibraryPageFragment<BookmarkNode>(), UserInteractionHan
             }
         }
     }
+
+    private suspend fun syncBookmarks() {
```

Comment:
As far as I can tell, this logic doesn't seem coupled to the fragment. Could it be moved into the bookmarks controller?

Reply:
Keeping it in the fragment makes testing a bit easier since it allows passing the function to the controller's constructor.

Reply:
If you just need to check if this function is called in the tests, I think it would be better to make it a `@VisibleForTesting` function in the controller and use a spy (and optionally mock it so it doesn't run any code in other tests).

Reply:
Ended up moving this logic to `handleRequestSync` in the controller. It was giving me a lot of trouble trying to test it so I just tested the order of sync related dispatches which is what I was more concerned with anyway.